Bug: learn-ocaml-client 0.14.0 can't build on win10 wsl2 debian due to ssl 0.5.10 build fails

**Related project scope(s):** …
learn-ocaml-client.0.14.0, ssl.0.5.10

## Bug description
ssl.0.5.10 won't build on wsl2 debian win10, thus cause learn-ocaml-client.0.14.0 fails to install.  
```
ty@sf10up:~$ opam install utop learn-ocaml-client
[NOTE] Package utop is already installed (current version is 2.9.2).
The following actions will be performed:
  ↘ downgrade ssl                0.5.12 to 0.5.10 [required by learn-ocaml-client]
  ∗ install   lwt_ssl            1.1.3            [required by learn-ocaml-client]
  ∗ install   conduit-lwt-unix   1.3.0            [required by cohttp-lwt-unix]
  ∗ install   cohttp-lwt-unix    4.0.0            [required by learn-ocaml-client]
  ∗ install   learn-ocaml-client 0.14.0
===== ∗ 4   ↘ 1 =====
Do you want to continue? [Y/n] Y

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⬇ retrieved cohttp-lwt-unix.4.0.0  (cached)
⬇ retrieved conduit-lwt-unix.1.3.0  (cached)
⬇ retrieved lwt_ssl.1.1.3  (cached)
⬇ retrieved learn-ocaml-client.0.14.0  (cached)
⬇ retrieved ssl.0.5.10  (cached)
[ERROR] The compilation of ssl.0.5.10 failed at "dune build -p ssl -j 3".

#=== ERROR while compiling ssl.0.5.10 =========================================#
# context     2.1.3 | linux/x86_64 | ocaml-base-compiler.4.12.0 | https://opam.ocaml.org#94fb49a6
# path        ~/.opam/4.12.0/.opam-switch/build/ssl.0.5.10
# command     ~/.opam/4.12.0/bin/dune build -p ssl -j 3
# exit-code   1
# env-file    ~/.opam/log/ssl-28030-12f498.env
# output-file ~/.opam/log/ssl-28030-12f498.out
### output ###
# /home/ty/.opam/4.12.0/lib/ocaml/caml/mlvalues.h:282:23: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
# [...]
#                        ^
# ssl_stubs.c:1179:18: note: in expansion of macro ‘String_val’
#    char *CApath = String_val(ca_path);
#                   ^~~~~~~~~~
# ssl_stubs.c: In function ‘ocaml_ssl_set_client_SNI_hostname’:
# /home/ty/.opam/4.12.0/lib/ocaml/caml/mlvalues.h:282:23: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
#  #define String_val(x) ((const char *) Bp_val(x))
#                        ^
# ssl_stubs.c:1262:20: note: in expansion of macro ‘String_val’
#    char *hostname = String_val(vhostname);
#                     ^~~~~~~~~~



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build ssl 0.5.10
└─
╶─ No changes have been performed
```

### To reproduce

Steps to reproduce the behavior:

On win10 wsl2 debian:  
`opam install learn-ocaml-client` or 
`opam install learn-ocaml-client.0.14.0` or 
`opam install ssl.0.5.10`

## Current configuration

<!-- You can write "N/A" for the items that sound irrelevant to you -->
* OS name (and version):  win10 21H2 19044.2006
* Browser name (and version): N/A
* `learn-ocaml --version`: 0.14.0
* `git describe --long --always --abbrev=40 --tags`: N/A

## Additional context
ssl 0.5.12 builds well, if in future can move dependency to this version, things would be great I think.

--- a/learn-ocaml-client.opam
+++ b/learn-ocaml-client.opam
@@ -42,7 +42,7 @@ depends: [
   "ppxlib"
   "ppx_sexp_conv"
   "ppx_tools"
-  "ssl" {= "0.5.10"}
+  "ssl" {= "0.5.12"}
   "vg"
 ]
 build: [

--- a/learn-ocaml-client.opam.locked
+++ b/learn-ocaml-client.opam.locked
@@ -95,7 +95,7 @@ depends: [
   "seq" {= "0.2.2"}
   "sexplib" {= "v0.14.0"}
   "sexplib0" {= "v0.14.0"}
-  "ssl" {= "0.5.10"}
+  "ssl" {= "0.5.12"}
   "stdlib-shims" {= "0.3.0"}
   "stringext" {= "1.6.0"}
   "topkg" {= "1.0.3"}

--- a/learn-ocaml.opam
+++ b/learn-ocaml.opam
@@ -59,7 +59,7 @@ depends: [
   "ppx_tools"
   "ppx_tools_versioned"
   "re"
-  "ssl" {= "0.5.10"}
+  "ssl" {= "0.5.12"}
   "uutf" {>= "1.0" }
   "vg"
   "yojson" {>= "1.4.0" }

--- a/learn-ocaml.opam.locked
+++ b/learn-ocaml.opam.locked
@@ -107,7 +107,7 @@ depends: [
   "seq" {= "0.2.2"}
   "sexplib" {= "v0.14.0"}
   "sexplib0" {= "v0.14.0"}
-  "ssl" {= "0.5.10"}
+  "ssl" {= "0.5.12"}
   "stdlib-shims" {= "0.3.0"}
   "stringext" {= "1.6.0"}
   "topkg" {= "1.0.3"}

